include/irr/builtin/material_compiler/glsl/common.glsl

```diff
@@ -1140,6 +1133,11 @@ irr_glsl_LightSample irr_bsdf_cos_generate(in MC_precomputed_t precomp, in instr
 		}
 		s = irr_glsl_createLightSampleTangentSpace(localV, localL, tangentFrame);
 		out_microfacet = irr_glsl_calcAnisotropicMicrofacetCache(currInteraction, s);
+
+		const float alpha = is_bsdf ? 0.0 : ax2;
+		const vec3 reflectance = is_bsdf ? vec3(1.0) : refl;
```

Comment:
whats the messing around with alpha and reflectance?

Reply:
why is the diffuse smooth when its a transmissive? and why is it not allowed an albedo?